trin-history/src/validation.rs

```diff
@@ -30,18 +29,36 @@ impl Validator<HistoryContentKey> for ChainHistoryValidator {
         content: &[u8],
     ) -> anyhow::Result<ValidationResult<HistoryContentKey>> {
         match content_key {
-            HistoryContentKey::BlockHeaderWithProof(key) => {
+            HistoryContentKey::BlockHeaderByHash(key) => {
                 let header_with_proof =
                     HeaderWithProof::from_ssz_bytes(content).map_err(|err| {
                         anyhow!("Header with proof content has invalid encoding: {err:?}")
                     })?;
                 let header_hash = header_with_proof.header.hash();
-                if header_hash != B256::from(key.block_hash) {
-                    return Err(anyhow!(
-                        "Content validation failed: Invalid header hash. Found: {header_hash:?} - Expected: {:?}",
-                        hex_encode(key.block_hash)
-                    ));
-                }
+                ensure!(
+                    header_hash == B256::from(key.block_hash),
+                    "Content validation failed: Invalid header hash. Found: {header_hash:?} - Expected: {:?}",
+                    hex_encode(header_hash)
+                );
+                self.header_oracle
+                    .read()
+                    .await
+                    .header_validator
+                    .validate_header_with_proof(&header_with_proof)?;
+
+                Ok(ValidationResult::new(true))
+            }
+            HistoryContentKey::BlockHeaderByNumber(key) => {
+                let header_with_proof =
+                    HeaderWithProof::from_ssz_bytes(content).map_err(|err| {
+                        anyhow!("Header with proof content has invalid encoding: {err:?}")
```

Comment:
Maybe it'd be useful to differentiate this error message from the one on line 35